src/ice.c

```diff
@@ -4078,24 +4078,30 @@ static void janus_ice_rtp_extension_update(janus_ice_handle *handle, janus_ice_p
 				}
 			}
 		}
+		/* Parse the abs-capture-time extension from source if needed */
+		if(handle->pc->abs_capture_time_ext_id != -1) {
+			uint64_t abs_ts = 0;
+			if(janus_rtp_header_extension_parse_abs_capture_time(packet->data, packet->length,
+					handle->pc->abs_capture_time_ext_id, &abs_ts) == 0) {
+				packet->extensions.abs_capture_ts = abs_ts;
+			}
+		}
 		/* Check if we need to add the abs-capture-time extension */
 		if(packet->extensions.abs_capture_ts > 0 && handle->pc->abs_capture_time_ext_id > 0) {
 			uint64_t abs64 = htonll(packet->extensions.abs_capture_ts);
 			if(!use_2byte) {
-				*index = (handle->pc->abs_capture_time_ext_id << 4) + 15;
+				*index = (handle->pc->abs_capture_time_ext_id << 4) + 7;
 				memcpy(index+1, &abs64, 8);
-				memset(index+9, 0, 8);
-				index += 17;
-				extlen += 17;
-				extbufsize -= 17;
+				index += 9;
+				extlen += 9;
+				extbufsize -= 9;
 			} else {
 				*index = handle->pc->abs_capture_time_ext_id;
-				*(index+1) = 16;
+				*(index+1) = 8;
 				memcpy(index+2, &abs64, 8);
-				memset(index+8, 0, 8);
-				index += 18;
-				extlen += 18;
-				extbufsize -= 18;
+				index += 10;
+				extlen += 10;
+				extbufsize -= 10;
```

Comment:
Not sure why you changed the block above? To my knowledge it wasn't broken. If it was please elaborate.